### ISSUE TITLE
client_id added to end session URL twice, but only in automated tests

### ISSUE THREAD

**Issue by mellis481** [authts/oidc-client-ts#1354]
I am currently (finally) switching from oidc-client-js to oidc-client-ts. It was a fairly easy migration with one exception: I'm seeing a failing Cypress test that validates the scenario of signing out due to inactivity.

In the test, when the signout occurs (which means calling `userManager.signoutRedirect()`), the following redirect is attempted:

```
https://auth.mycompany.net/v2/logout?
client_id=MY_APPLICATION_ID&
client_id=MY_APPLICATION_ID&
returnTo=https://localhost:9000/ui/loggedout&
post_logout_redirect_uri=https://localhost:9000/ui/loggedout?timeout=true
```

As you can see, `client_id` is added to the logout redirect URL query params twice. This results in our IdP (Auth0) throwing a _The client id is malformed, string type expected_ error (since multiple query params with the same key is viewed as an array).

Here are the client settings that are being used in my app's development environment:

```
{
  "authority": "https://auth.mycompany.net",
  "automaticSilentRenew": true,
  "client_id": "MY_APPLICATION_ID",
  "response_type": "code",
  "redirect_uri": "https://www.mycompany.net/ui/authorize",
  "scope": "openid profile offline_access",
  "post_logout_redirect_uri": "https://www.mycompany.net/ui/loggedout",
  "extraQueryParams": {
    "audience": "mycompany://migration-api"
  },
  "metadata": {
    "authorization_endpoint": "https://auth.mycompany.net/authorize",
    "end_session_endpoint": "https://auth.mycompany.net/v2/logout?client_id=MY_APPLICATION_ID&returnTo=https://www.mycompany.net/ui/loggedout",
    "issuer": "https://auth.mycompany.net/",
    "jwks_uri": "https://auth.mycompany.net/.well-known/jwks.json",
    "userinfo_endpoint": "https://auth.mycompany.net/userinfo",
    "token_endpoint": "https://auth.mycompany.net/oauth/token"
  }
}
```

This was not changed since we used oidc-client-js.  One thing I will point out about these settings is that, when we originally implemented this solution with oidc-client-js, we found that we needed to include `client_id` in the `metadata.end_session_endpoint` value or else the `client_id` would not be included in the end session URL. It seems like others have run into this too ([here](https://github.com/authts/oidc-client-ts/issues/802) is an example). If I remove the `client_id` from the `metadata.end_session_endpoint` value after integrating with oidc-client-ts, my Cypress test will run successfully meaning `client_id` is only included once in the end session URL. The problem, however, is that in the actual running application (where everything works fine), the end session URL does not have any `client_id` (after removing `client_id` from the `metadata.end_session_endpoint` value).

So something is causing `client_id` to be added to the end session URL twice, but only in our Cypress test. Any thoughts as to why this is happening?

**Comment by pamapa** [comment]
The second `client_id` comes most probably from https://github.com/authts/oidc-client-ts/blob/a4eb61e4dde7fb626a7fb11ec2753524ed887ab7/src/OidcClient.ts#L250

The question is why? Can you debug that, verify is that is true and provide a fix?

**Comment by mellis481** [comment]
@pamapa Thanks for pointing me to this part of the code. It turns out, `id_token_hint` was `undefined` in my Cypress test, but not in my application. My Cypress test is a mocked-API-style test where we create a fake token rather than an e2e test where an actual login occurs. The fake session storage object I was creating has an `access_token` prop (which is a simple, fake, valid JWT), but not `id_token`. After adding an `id_token` prop (the same as the `access_token` prop), it works. 

Thanks again!

**Comment by kevingentile** [comment]
@pamapa  I encountered this issue in my application. The signout redirect included two instance of the query parameter `client_id`. Auth0 misinterpreted this duplicate and reported an error: "The client id is malformed, string type expected". To fix, I consulted the [documentation](https://auth0.com/docs/authenticate/login/logout/log-users-out-of-auth0#oidc-logout-endpoint-parameters) and provided the `id_token_hint` to `signoutRedirect`:
```
const hint = await this.manager?.getUser().then((u) => u?.id_token);
...
await this.manager?.signoutRedirect({id_token_hint: hint })
```

It's unclear if my original approach, calling only `signoutRedirect()`, was a misconfiguration that the client might have identified and logged a warning. Is there a scenario where including a duplicate `client_id` query parameter in the redirect would be valid?
